tests/utils.go

```diff
@@ -423,6 +440,183 @@ func CreateSecret(name string, data map[string]string) {
+	var singleYamlDocument []byte
+	for {
+		n, err := yamlDocDecoder.Read(buff)
+		// Copy buffer into single yaml document variable
+		singleYamlDocument = append(singleYamlDocument, buff[:n]...)
+		if err == io.ErrShortBuffer {
+			// Buffer is short to read whole document, reading again
+			continue
+		} else if err == io.EOF {
+			// There is nothing else to read
+			break
+		}
+		// Unexpected error
+		PanicOnError(err)
+		// Got full single yaml document
+		jsonrepr, err := yaml.ToJSON(singleYamlDocument)
+		PanicOnError(err)
+		// Clear variable for single yaml document
+		singleYamlDocument = singleYamlDocument[:0]
+		var obj map[string]interface{}
+		err = json.Unmarshal([]byte(jsonrepr), &obj)
+		if err != nil {
+			fmt.Printf(fmt.Sprintf("ERROR: Can not unmarshall YAML %s\n", err))
+			continue
+		}
+		unstruct := unstructured.Unstructured{}
+		unstruct.SetUnstructuredContent(obj)
+		if unstruct.GetKind() == "" {
+			// Skip empty structss
+			continue
+		}
+		objects = append(objects, unstruct)
+	}
+	return objects
+}
+
+func isNamespaceScoped(kind schema.GroupVersionKind) bool {
+	switch kind.Kind {
+	case "ClusterRole", "ClusterRoleBinding":
+		return false
+	}
+	return true
+}
+
+func IsOpenShift() bool {
+	virtClient, err := kubecli.GetKubevirtClient()
+	PanicOnError(err)
+
+	result := virtClient.RestClient().Get().AbsPath("/version/openshift").Do()
+
+	if result.Error() == nil {
+		return true
+	}
+	var code int
+	result.StatusCode(&code)
+	if code != http.StatusNotFound {
+		fmt.Printf(fmt.Sprintf("ERROR: Can not determine cluster type %#v\n", result))
+		panic(err)
+	}
+	return false
+}
+
+func composeResourceURI(object unstructured.Unstructured) string {
+	uri := "/api"
+	if object.GetAPIVersion() != "v1" {
+		uri += "s"
+	}
+	uri += "/" + object.GetAPIVersion()
+	if object.GetNamespace() != "" && isNamespaceScoped(object.GroupVersionKind()) {
+		uri += "/namespaces/" + object.GetNamespace()
+	}
+	uri += "/" + strings.ToLower(object.GetKind())
+	if object.GetKind()[len(object.GetKind())-1:] != "s" {
+		uri += "s"
+	}
+	return uri
+}
+
+func ApplyRawManifest(object unstructured.Unstructured) error {
+	virtCli, err := kubecli.GetKubevirtClient()
+	PanicOnError(err)
+
+	uri := composeResourceURI(object)
+	jsonbody, err := object.MarshalJSON()
+	PanicOnError(err)
+	b, err := virtCli.CoreV1().RESTClient().Post().RequestURI(uri).Body(jsonbody).DoRaw()
+	if err != nil {
+		fmt.Printf(fmt.Sprintf("ERROR: Can not apply %s\n", object))
+		panic(err)
+	}
+	status := unstructured.Unstructured{}
+	return json.Unmarshal(b, &status)
+}
+
+func DeleteRawManifest(object unstructured.Unstructured) error {
+	virtCli, err := kubecli.GetKubevirtClient()
+	PanicOnError(err)
+
+	uri := composeResourceURI(object)
+	uri = uri + "/" + object.GetName()
+	PanicOnError(err)
+	result := virtCli.CoreV1().RESTClient().Delete().RequestURI(uri).Do()
+	var code int
+	result.StatusCode(&code)
+	if result.Error() != nil && code != http.StatusNotFound {
```

Comment:
Did not think about this myself, but `result.Error() != nil && !errors.IsNotFound(err)` should have the same effect.